cosmos/operators/azure_container_instance.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
+
+from typing import Any, Callable, Sequence
+
+import yaml
+from airflow.utils.context import Context
+from cosmos.config import ProfileConfig
+
+from cosmos.log import get_logger
+from cosmos.operators.base import DbtBaseOperator
+
+logger = get_logger(__name__)
+
+# ACI is an optional dependency, so we need to check if it's installed
+try:
+    from airflow.providers.microsoft.azure.operators.container_instances import AzureContainerInstancesOperator
+except ImportError:
+    raise ImportError(
+        "Could not import AzureContainerInstancesOperator. Ensure you've installed the Microsoft Azure provider "
+        "separately or with `pip install astronomer-cosmos[...,azure-container-instance]`."
+    )
+
+
+class DbtAzureContainerInstanceBaseOperator(AzureContainerInstancesOperator, DbtBaseOperator):  # type: ignore
+    """
+    Executes a dbt core cli command in a Docker container.
+    """
+
+    template_fields: Sequence[str] = tuple(
+        list(DbtBaseOperator.template_fields) + list(AzureContainerInstancesOperator.template_fields)
+    )
+
+    intercept_flag = False
+
+    def __init__(
+        self,
+        ci_conn_id: str,
+        resource_group: str,
+        name: str,
+        image: str,
+        region: str,
+        profile_config: ProfileConfig | None = None,
+        remove_on_error: bool = False,
+        fail_if_exists: bool = False,
+        registry_conn_id: str | None = None,  # need to add a default in order for Airflow 2.3 support
+        **kwargs: Any,
+    ) -> None:
+        self.profile_config = profile_config
+        super().__init__(
+            ci_conn_id=ci_conn_id,
+            resource_group=resource_group,
+            name=name,
+            image=image,
+            region=region,
+            remove_on_error=remove_on_error,
+            fail_if_exists=fail_if_exists,
+            registry_conn_id=registry_conn_id,
+            **kwargs,
+        )
+
+    def build_and_run_cmd(self, context: Context, cmd_flags: list[str] | None = None) -> int:
+        self.build_command(context, cmd_flags)
+        self.log.info(f"Running command: {self.command}")
+        result = int(super().execute(context))
+        logger.info(result)
+        return result
+
+    def build_command(self, context: Context, cmd_flags: list[str] | None = None) -> None:
+        # For the first round, we're going to assume that the command is dbt
+        # This means that we don't have openlineage support, but we will create a ticket
+        # to add that in the future
+        self.dbt_executable_path = "dbt"
+        dbt_cmd, env_vars = self.build_cmd(context=context, cmd_flags=cmd_flags)
```

Comment:
Is it possible to set environment variables for Azure container instances? Browsing the operator [here](https://airflow.apache.org/docs/apache-airflow-providers-microsoft-azure/stable/_modules/airflow/providers/microsoft/azure/operators/container_instances.html#AzureContainerInstancesOperator), it looks like `self.environment_variables` set to `env_vars` here would work.

Reply:
Indeed, I've changed this now